mqtt/mqtt-broker/src/persist.rs

```diff
@@ -313,14 +314,32 @@ where
                         path.clone(),
                     ))))
                 });
-                let file = OpenOptions::new()
+                let mut file = OpenOptions::new()
                     .read(true)
                     .open(&path)
                     .context(ErrorKind::Persist(ErrorReason::FileOpen(path)))?;
 
                 fail_point!("filepersistor.load.format", |_| {
                     Err(Error::from(ErrorKind::Persist(ErrorReason::Serialize)))
                 });
+
+                debug!("reading version");
+                let mut size = [0; std::mem::size_of::<u32>()];
+                file.read_exact(&mut size)
+                    .map_err(|_| Error::from(ErrorKind::Persist(ErrorReason::ReadVersion)))?;
+                let size: usize = u32::from_be_bytes(size).try_into().unwrap();
+
+                let mut version = Vec::with_capacity(size);
+                unsafe {
```

Comment:
I think we should avoid unsafe code as much as we can. Especially in such simple case

Reply:
The only safe way to do this I could find fills every value with 0, which seems inefficient. Is there a better way to get a mutable slice of a specified size?

Reply:
I'll just use the safe func that fills it with 0. It's slightly inefficient, but its very minor